### PR TITLE
WIP | Update Unit test in SettingResourceTest to test the API endpoin…

### PR DESCRIPTION
**Pull request by rehammuzzamil**
…ts instead of the service

Issue # 207
Update Unit test in SettingResourceTest to test the API endpoints instead of the service

### CODE REVIEW

File: src/test/java/org/opensrp/web/rest/SettingResourceTest.java [hunk 1]
```diff
@@ -116,81 +137,47 @@ public void setUp() {
...
-	
+
+	@Test
+	public void testPostSaveSetting() throws Exception {
+		String SETTINGS_JSON = "{\"settingConfigurations\":[]}";
+
+		when(settingService.saveSetting(Matchers.any(String.class))).thenReturn("ID-12345");
+		MvcResult mvcResult = this.mockMvc.perform(
+				post(BASE_URL + "/sync").contentType(MediaType.APPLICATION_JSON).body(SETTINGS_JSON.getBytes()).accept(MediaType.APPLICATION_JSON))
+				.andExpect(status().isOk()).andReturn();
```
Comment (githengi) [inline src/test/java/org/opensrp/web/rest/SettingResourceTest.java:176]: assert the service class was called with the correct params

**Comment by rehammuzzamil** [comment]
> The unit for `testFindSettingsByVersionAndTeamId` has been removed

Do we need to include that test as well?

**Comment by githengi** [comment]
Yes all tests need to be retained.

File: src/test/java/org/opensrp/web/rest/SettingResourceTest.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/test/java/org/opensrp/web/rest/SettingResourceTest.java:160]: Done.

**Comment by rehammuzzamil** [comment]
> Yes all tests need to be retained.



> The unit for `testFindSettingsByVersionAndTeamId` has been removed

Added those tests again.

**Comment by githengi** [comment]
@rehammuzzamil fix the issues Identified by Codacy and update the branch with master

**Comment by rehammuzzamil** [comment]
> @rehammuzzamil fix the issues Identified by Codacy and update the branch with master

Done

**Pull request merged**